pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -309,14 +310,35 @@ func findContainerIDByName(status *v1.PodStatus, name string) (string, error) {
 	return "", fmt.Errorf("unable to find ID for container with name %v in pod status (it may not be running)", name)
 }
 
-func (m *manager) updateContainerCPUSet(containerID string, cpus cpuset.CPUSet) error {
+func (m *manager) updateContainerCPUSet(pod *v1.Pod, container *v1.Container, containerID string, cpus cpuset.CPUSet) error {
```

Comment:
isnt cfs quota still going to be applied at the pod cgroup with this change?

see:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/pod_container_manager_linux.go#L87
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/helpers_linux.go#L164

Reply:
Yes, unfortunately it is. Running in [kind](https://kind.sigs.k8s.io/) gives me:
```
> cat /sys/fs/cgroup/cpu,cpuacct/docker/8d89c6f3cf8e5aaace1beb1adad3f38b926420ff5490d53a80a18b4457a60834/kubepods/pod71ef1b9f-8373-4dd8-9d57-a47706112750/cpu.cfs_quota_us
100000
> cat /sys/fs/cgroup/cpu,cpuacct/docker/8d89c6f3cf8e5aaace1beb1adad3f38b926420ff5490d53a80a18b4457a60834/kubepods/pod71ef1b9f-8373-4dd8-9d57-a47706112750/7f21da8aaab220ea945cca3c29c8af1217a09d2b146721b2f98b051d063c74a5/cpu.cfs_quota_us
-1
```

I checked `cpu.stat` for the container and it had `throttled_time 0`, but `cpu.stat` for the pod (one level up in the hierarchy) did have a non-zero `throttled_time` so there is some throttling going on.

Solving this is a little more complicated. Unsetting the CFS CPU quota on the pod level would always make sense if there is only one container running with static CPU policies assigned, but with multiple containers in a pod we would have to check whether that holds for all of them. Such a check could placed in the `cpu_manager`'s `reconcileState` function, but I'm not sure whether the pod resources can be easily updated. Do you know whether that is the case?